File: src/MonitoringPlatform.Domain/Entities/Monitor.cs
using System;

namespace MonitoringPlatform.Domain.Entities
{

    public class Monitor
    {
        public Guid Id { get; set; }
        public Guid TenantId { get; set; }
        public string Name { get; set; } = null!;
        public string Url { get; set; } = null!;
        public int IntervalInSeconds { get; set; }
        public MonitorStatus Status { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

        public Tenant Tenant { get; set; } = null!;
        public ICollection<MonitorLog> Logs { get; set; } = new List<MonitorLog>();
    }

    public enum MonitorStatus
    {
        Unknown = 0,
        Online = 1,
        Offline = 2
    }
}

File: frontend/src/types/monitor.ts
export const MonitorStatus = {
  Unknown: 0,
  Online: 1,
  Offline: 2,
} as const;

export type MonitorStatus = (typeof MonitorStatus)[keyof typeof MonitorStatus];

export type MonitorResponseDto = {
  id: string;
  name: string;
  url: string;
  intervalInSeconds: number;
  status: MonitorStatus;
  createdAt: string;
  updatedAt: string;
};

export type CreateMonitorDto = {
  name: string;
  url: string;
  intervalInSeconds: number;
};

export type UpdateMonitorDto = CreateMonitorDto;

export function monitorStatusLabel(status: MonitorStatus): string {
  switch (status) {
    case MonitorStatus.Online:
      return "Online";
    case MonitorStatus.Offline:
      return "Offline";
    case MonitorStatus.Unknown:
    default:
      return "Unknown";
  }
}
